borrow-collateral-ratio/src/agent.js

```diff
@@ -0,0 +1,146 @@
+const {
+  Finding, FindingSeverity, FindingType, ethers, getEthersProvider,
+} = require('forta-agent');
+const BigNumber = require('bignumber.js');
+
+const { getAbi } = require('./utils');
+
+// load any bot configuration parameters
+const config = require('../bot-config.json');
+
+// set up a variable to hold initialization data used in the handler
+const initializeData = {};
+const DECIMALS_ABI = ['function decimals() view returns (uint)'];
+
+function createAlert(protocolName, protocolAbbrev, developerAbbrev, type, severity, metadata) {
+  const { assetTokenSymbol, assetTokenAddress, maxUtilizationRate } = metadata;
+  return Finding.fromObject({
+    name: `${protocolName} Borrow Collateral Ratio`,
+    description: 'The ratio of total borrow amount to total liquidity exceeds the configured '
+      + `threshold of ${maxUtilizationRate}% for asset ${assetTokenAddress} (${assetTokenSymbol})`,
+    alertId: `${developerAbbrev}-${protocolAbbrev}-BORROW-COLLATERAL-RATIO`,
+    type: FindingType[type],
+    severity: FindingSeverity[severity],
+    protocol: protocolName,
+    metadata: {
+      ...metadata,
+    },
+  });
+}
+
+function provideInitialize(data) {
+  return async function initialize() {
+    /* eslint-disable no-param-reassign */
+    // assign configurable fields
+    data.protocolName = config.protocolName;
+    data.protocolAbbreviation = config.protocolAbbreviation;
+    data.developerAbbreviation = config.developerAbbreviation;
+
+    const { maxUtilizationRate, type, severity } = config.ratioSettings;
+    data.maxUtilizationRate = maxUtilizationRate;
+    data.type = type;
+    data.severity = severity;
+
+    const provider = getEthersProvider();
+    const { ProtocolDataProvider: protocolDataProvider } = config.contracts;
+
+    const { address: protocolDataProviderAddress } = protocolDataProvider;
+    const protocolDataProviderAbi = getAbi(protocolDataProvider.abiFile);
```

Comment:
I think the code, as written, works just fine.  I just wanted to offer a suggestion for potentially doing all of the destructuring up front.

I can see an argument for doing it either way (i.e. doing the destructuring right next to where each value is used versus doing the destructuring up front and using the values later).
```suggestion
    const {
      protocolName,
      protocolAbbreviation,
      developerAbbreviation,
      ratioSettings: {
        maxUtilizationRate,
        type,
        severity
      },
      contracts: {
        ProtocolDataProvider: {
          address: protocolDataProviderAddress,
          abiFile,
        },
      },
    } = config;
    
    data.protocolName = protocolName;
    data.protocolAbbreviation = protocolAbbreviation;
    data.developerAbbreviation = developerAbbreviation;
    data.maxUtilizationRate = maxUtilizationRate;
    data.type = type;
    data.severity = severity;

    const provider = getEthersProvider();
    const protocolDataProviderAbi = getAbi(abiFile);
```

Reply:
In most cases I like destructuring objects at the time when I am going to use the value; I think it just makes the code easier to follow.